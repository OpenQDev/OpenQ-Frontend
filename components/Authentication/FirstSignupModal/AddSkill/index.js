import React, { useContext } from 'react';
import StoreContext from '../../../../store/Store/StoreContext';
import Cross from '../../../svg/cross';

const AddSkill = ({ category, rolesInCategoriesState }) => {
  const [rolesInCategories, setRolesInCategories] = rolesInCategoriesState;
  const theseRoles = rolesInCategories[category];
  const [appState, appDispatch] = useContext(StoreContext);
  const { accountData } = appState;

  const updateApiAndState = async (newRoles, category) => {
    const newRolesAndCategory = { ...rolesInCategories, [category]: newRoles };
    const userValues = accountData.github
      ? {
          github: accountData.github,
          ...newRolesAndCategory,
        }
      : {
          email: accountData.email,
          ...newRolesAndCategory,
        };
    const { updateUser } = await appState.openQPrismaClient.updateUser(userValues);
    appDispatch({ type: 'UPDATE_ACCOUNT_DATA', payload: updateUser });
    setRolesInCategories(newRolesAndCategory);
  };

  const removeRole = (e, removedRole) => {
    e.preventDefault();
    const newRoles = theseRoles.filter((role) => role !== removedRole);
    updateApiAndState(newRoles, category);
  };
  return (
    <>
      {theseRoles?.map((role, index) => {
        return (
          <button
            className={`flex items-center gap-2 py-0.5 btn-default w-fit m-1`}
            key={index}
            onClick={(e) => removeRole(e, role)}
          >
            {role}
            <Cross />
          </button>
        );
      })}
    </>
  );
};

export default AddSkill;
