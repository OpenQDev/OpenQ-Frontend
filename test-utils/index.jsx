// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/FundStore/TestFundProvider';
import TokenProvider from '../components/TokenSelection/TokenStore/TokenProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
afterEach(() => {
  cleanup();
});

const customRender = (ui, options = {}, storeProps, authProps) => {
  const IntersectionObserverMock = vi.fn(() => ({
    disconnect: vi.fn(),
    observe: vi.fn(),
    takeRecords: vi.fn(),
    unobserve: vi.fn(),
  }));
  vi.stubGlobal('IntersectionObserver', IntersectionObserverMock);
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <TokenProvider>{children}</TokenProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}>
            <TokenProvider>{children}</TokenProvider>
          </FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
