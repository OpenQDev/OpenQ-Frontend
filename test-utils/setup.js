import { expect, beforeEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import matchers from '@testing-library/jest-dom/matchers';

expect.extend(matchers);
beforeEach(() => {
  cleanup();
});
process.on('unhandledRejection', (reason) => {
  // eslint-disable-next-line no-console
  console.log(`FAILED TO HANDLE PROMISE REJECTION`);
  throw reason;
});
