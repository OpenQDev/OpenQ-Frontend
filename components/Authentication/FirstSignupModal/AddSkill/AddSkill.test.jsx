import { render, screen } from '../../../../test-utils';
import React from 'react';
import AddSkill from '.';

describe('Add Skill', () => {
  const user = {
    id: '63c58870f2943bf006029305',
    renderError: '',
    avatarUrl: 'https://avatars.githubusercontent.com/u/75732239?u=285a48b76fb74f9487a3f595002bdbd69522a151&v=4',
    __typename: 'User',
    name: 'Anya',
    login: 'ArcAnya',
    websiteUrl: 'arcanya.com',
    url: 'https://github.com/ArcAnya',
    twitterUsername: null,
    recentLanguages: ['Dockerfile', 'JavaScript', 'CSS', 'TypeScript', 'Mustache'],
    onChainAddress: null,
    github: 'MDQ6VXNlcjc1NzMyMjM5',
    devRoles: [],
    otherRoles: ['ohia'],
    interests: [],
    languages: ['dockerfile', 'javascript', 'css', 'typescript', 'mustache', 'rust'],
    frameworks: [],
  };
  it('should match initial DOM Snapshot', () => {
    const setRolesInCategoriesState = vi.fn();
    const rolesInCategoriesState = [{ languages: user.languages }, setRolesInCategoriesState];
    const { asFragment } = render(
      <AddSkill
        rolesInCategoriesState={rolesInCategoriesState}
        childInfo={['toml', 'languages']}
        user={user}
        category={'languages'}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('setInputValue to have been called', async () => {
    const setRolesInCategoriesState = vi.fn();
    const rolesInCategoriesState = [{ languages: user.languages }, setRolesInCategoriesState];
    render(<AddSkill rolesInCategoriesState={rolesInCategoriesState} category={'languages'} />);
    const buttons = screen.getAllByRole('button');
    expect(buttons.length).toBe(6);
  });
});
