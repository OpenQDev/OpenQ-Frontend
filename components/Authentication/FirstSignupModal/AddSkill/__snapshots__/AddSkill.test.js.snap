// vi Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Skill should match initial DOM Snapshot 1`] = `
<DocumentFragment>
  <button
    class="flex items-center gap-2 py-0.5 btn-default w-fit m-1"
  >
    dockerfile
    <svg
      class="fill-muted hover:fill-blue-400 w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <button
    class="flex items-center gap-2 py-0.5 btn-default w-fit m-1"
  >
    javascript
    <svg
      class="fill-muted hover:fill-blue-400 w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <button
    class="flex items-center gap-2 py-0.5 btn-default w-fit m-1"
  >
    css
    <svg
      class="fill-muted hover:fill-blue-400 w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <button
    class="flex items-center gap-2 py-0.5 btn-default w-fit m-1"
  >
    typescript
    <svg
      class="fill-muted hover:fill-blue-400 w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <button
    class="flex items-center gap-2 py-0.5 btn-default w-fit m-1"
  >
    mustache
    <svg
      class="fill-muted hover:fill-blue-400 w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <button
    class="flex items-center gap-2 py-0.5 btn-default w-fit m-1"
  >
    rust
    <svg
      class="fill-muted hover:fill-blue-400 w-4 h-4"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;
